src/data_source/connected.rs

```diff
@@ -3,13 +3,37 @@ use std::mem::forget;
 use std::ops::Deref;
 use std::ptr;
 use std::thread::panicking;
+use std::marker::PhantomData;
+
+/// State used by `Connected`. Means that autocommit is enabled
+#[derive(Debug)]
+#[allow(missing_copy_implementations)]
+pub enum AutocommitOn {}
+
+/// State used by `Connected`. Means that autocommit is disabled
+#[derive(Debug)]
+#[allow(missing_copy_implementations)]
+pub enum AutocommitOff {}
+
+/// Marker trait for autocommit mode state types
+pub trait AutocommitMode {}
+
+impl AutocommitMode for AutocommitOn {}
+impl AutocommitMode for AutocommitOff {}
 
 /// An `HDbc` with the additional invariant of being 'connected'.
 #[derive(Debug)]
-pub struct Connected<'env>(HDbc<'env>);
+pub struct Connected<'env, AC: AutocommitMode>(HDbc<'env>, PhantomData<AC>);
 
-impl<'env> Drop for Connected<'env> {
+impl<'env, AC: AutocommitMode> Drop for Connected<'env, AC> {
     fn drop(&mut self) {
+        match self.0.rollback() {
```

Comment:
I am not sure this is necessary, wont disconnecting do this anyways?

Reply:
If you'll try to disconnect while transaction is in progress ODBC will return error, so IMO it's better to rollback implicitly

Reply:
I agree. (I probably never checked disconnect results since both rollbacks and disconnect often happen in destructor.)

Reply:
Now I think it's better to move this part to odbc-rs leaving this crate ad close to ODBC as possible